vms/platformvm/txs/fee/complexity.go

```diff
@@ -0,0 +1,594 @@
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/fee"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/platformvm/fx"
+	"github.com/ava-labs/avalanchego/vms/platformvm/signer"
+	"github.com/ava-labs/avalanchego/vms/platformvm/stakeable"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+const (
+	intrinsicValidatorBandwidth = ids.NodeIDLen + // nodeID
+		wrappers.LongLen + // start
+		wrappers.LongLen + // end
+		wrappers.LongLen // weight
+
+	intrinsicSubnetValidatorBandwidth = intrinsicValidatorBandwidth + // validator
+		ids.IDLen // subnetID
+
+	intrinsicOutputBandwidth = ids.IDLen + // assetID
+		wrappers.IntLen // output typeID
+
+	intrinsicStakeableLockedOutputBandwidth = wrappers.LongLen + // locktime
+		wrappers.IntLen // output typeID
+
+	intrinsicSECP256k1FxOutputOwnersBandwidth = wrappers.LongLen + // locktime
+		wrappers.IntLen + // threshold
+		wrappers.IntLen // num addresses
+
+	intrinsicSECP256k1FxOutputBandwidth = wrappers.LongLen + // amount
+		intrinsicSECP256k1FxOutputOwnersBandwidth
+
+	intrinsicInputBandwidth = ids.IDLen + // txID
+		wrappers.IntLen + // output index
+		ids.IDLen + // assetID
+		wrappers.IntLen + // input typeID
+		wrappers.IntLen // credential typeID
+
+	intrinsicStakeableLockedInputBandwidth = wrappers.LongLen + // locktime
+		wrappers.IntLen // input typeID
+
+	intrinsicSECP256k1FxInputBandwidth = wrappers.IntLen + // num indices
+		wrappers.IntLen // num signatures
+
+	intrinsicSECP256k1FxTransferableInputBandwidth = wrappers.LongLen + // amount
+		intrinsicSECP256k1FxInputBandwidth
+
+	intrinsicSECP256k1FxSignatureBandwidth = wrappers.IntLen + // signature index
+		secp256k1.SignatureLen // signature length
+
+	intrinsicPoPBandwidth = bls.PublicKeyLen + // public key
+		bls.SignatureLen // signature
+)
+
+var (
+	_ txs.Visitor = (*complexityVisitor)(nil)
+
+	IntrinsicAddPermissionlessValidatorTxComplexities = fee.Dimensions{
+		fee.Bandwidth: IntrinsicBaseTxComplexities[fee.Bandwidth] +
+			intrinsicValidatorBandwidth + // validator
+			ids.IDLen + // subnetID
+			wrappers.IntLen + // signer typeID
+			wrappers.IntLen + // num stake outs
+			wrappers.IntLen + // validator rewards typeID
+			wrappers.IntLen + // delegator rewards typeID
+			wrappers.IntLen, // delegation shares
+		fee.DBRead:  1,
+		fee.DBWrite: 1,
+		fee.Compute: 0,
+	}
+	IntrinsicAddPermissionlessDelegatorTxComplexities = fee.Dimensions{
+		fee.Bandwidth: IntrinsicBaseTxComplexities[fee.Bandwidth] +
+			intrinsicValidatorBandwidth + // validator
+			ids.IDLen + // subnetID
+			wrappers.IntLen + // num stake outs
+			wrappers.IntLen, // delegator rewards typeID
+		fee.DBRead:  1,
+		fee.DBWrite: 0,
```

Comment:
can you clarify why this is 0, while for validators (both primary network and subnets) it is 1?

Reply:
This was incorrect